### ISSUE TITLE
`@GuardedBy("itself")` is accepted but not checked

### ISSUE THREAD

**Issue by Stephan202** [google/error-prone#541]
When annotating a field `@GuardedBy("somenonexistentfield")` this fails with an error, as it should:

> `Invalid @GuardedBy expression: could not resolve guard`

But the annotation `@GuardedBy("itself")` is accepted, yet not enforced. This leads to a false sense of security. See the example code below:

```java
public final class GuardedByTest {
    @GuardedBy("itself")
    private final List<String> list1 = new ArrayList<>();

    @GuardedBy("list2")
    private final List<String> list2 = new ArrayList<>();

    public void regularLoops() {
        // This should trigger a warning but doesn't.
        for (int i = 0; i < 1; i++) {
            this.list1.add(String.valueOf(i));
        }

        // This is correct; no warning.
        synchronized (this.list2) {
            for (int i = 0; i < 1; i++) {
                this.list2.add(String.valueOf(i));
            }
        }
    }

    public void streamLoopsWithMethodHandles() {
        // This should trigger a warning but doesn't.
        IntStream.range(0, 1).mapToObj(String::valueOf).forEach(this.list1::add);

        // This is correct; no warning.
        synchronized (this.list2) {
            IntStream.range(0, 1).mapToObj(String::valueOf).forEach(this.list2::add);
        }
    }

    public void streamLoopsWithLambda() {
        // This should trigger a warning but doesn't.
        IntStream.range(0, 1).forEach(i -> this.list1.add(String.valueOf(i)));

        // False positive triggered. See google/error-prone#536.
        synchronized (this.list2) {
            IntStream.range(0, 1).forEach(i -> this.list2.add(String.valueOf(i)));
            //                                     ^
            // error: [GuardedByChecker] This access should be guarded by 'this.list2', which is not currently held
        }
    }
}
```
